### PR TITLE
api_proto_plugin: create visitor per file

### LINKED ISSUE #9266: TypeWhisperer generates incorrect proto path and type_db_gen depends on order
For example, `bazel-bin/external/envoy_api/envoy/admin/v2alpha/pkg/envoy/admin/v2alpha/tap.proto.types.pb_text` contains much more than what in envoy/admin/v2alpha/tap.proto:

```
types {
  key: "envoy.admin.v2alpha.BootstrapConfigDump"
  value {
    qualified_package: "envoy.admin.v2alpha"
    proto_path: "envoy/admin/v2alpha/tap.proto"
    type_dependencies: "google.protobuf.Timestamp"
    type_dependencies: "envoy.config.bootstrap.v2.Bootstrap"
  }
}
types {
  key: "envoy.admin.v2alpha.Certificate"
  value {
    qualified_package: "envoy.admin.v2alpha"
    proto_path: "envoy/admin/v2alpha/tap.proto"
    type_dependencies: "envoy.admin.v2alpha.CertificateDetails"
  }
}
types {
  key: "envoy.admin.v2alpha.CertificateDetails"
  value {
    qualified_package: "envoy.admin.v2alpha"
    proto_path: "envoy/admin/v2alpha/tap.proto"
    type_dependencies: "google.protobuf.Timestamp"
    type_dependencies: "envoy.admin.v2alpha.SubjectAlternateName"
  }
}
types {
  key: "envoy.admin.v2alpha.Certificates"
  value {
    qualified_package: "envoy.admin.v2alpha"
...
```

which should only include:
```
types {
  key: "envoy.admin.v2alpha.TapRequest"
  value {
    qualified_package: "envoy.admin.v2alpha"
    proto_path: "envoy/admin/v2alpha/tap.proto"
    type_dependencies: "envoy.service.tap.v2alpha.TapConfig"
  }
}
```

This seems because api_proto_plugin's visitor.Visitor should be stateless while TypeWhisperer is stateful and keeps types in self._types.

@htuch

### PR DESCRIPTION
**Pull request by lizan**
Signed-off-by: Lizan Zhou <lizan@tetrate.io>

Description:
Create visitor.Visitor per proto file to get correct information from TypeWhisperer.
This will make it easier to track imports in protoxform as well.

Risk Level: Low
Testing: CI
Docs Changes:
Release Notes:
Fixes #9266

### CODE REVIEW

File: tools/api_proto_plugin/plugin.py [hunk 1]
```diff
@@ -17,9 +17,9 @@
         # Output files are generated alongside their corresponding input .proto,
         # with the output_suffix appended.
         'output_suffix',
-        # The visitor is a visitor.Visitor defining the business logic of the plugin
-        # for the specific output descriptor.
-        'visitor',
+        # The visitor is a function to create a visitor.Visitor defining the business
```
Comment (htuch) [inline tools/api_proto_plugin/plugin.py]: Nit: "The visitor factory is a function.."

**Comment by htuch** [comment]
@lizan another master merge needed :)

**Comment by lizan** [comment]
@htuch this is now only doc fix because #9276 contains this one.

**Pull request merged**